File: app/src/main/java/com/example/tipcalculator/Bill.java
package com.example.tipcalculator;

public class Bill {

    private double amount, tipAmount, totalAmount, amountPerPerson;
    private int tip_percent, noPeople;

    public Bill(double amount, int tip_percent, int noPeople) throws ZeroAmountException {
        if(amount==0)
            throw new ZeroAmountException();

        this.amount = amount;
        this.tip_percent = tip_percent;
        this.noPeople = noPeople;
        calculate();
    }

    private void calculate() {
        tipAmount = tip_percent * amount / 100.0;
        totalAmount = tipAmount + amount;
        amountPerPerson = totalAmount / noPeople;
    }

    public double getTipAmount() {
        return tipAmount;
    }

    public double getTotalAmount() {
        return totalAmount;
    }

    public double getAmountPerPerson() {
        return amountPerPerson;
    }

    public class ZeroAmountException extends Exception {
    }
}

File: app/src/main/java/com/example/tipcalculator/Bill.kt
package com.example.tipcalculator

class Bill(amount: Double, tip_percent: Int, noPeople: Int) {
    private val amount: Double
    var tipAmount = 0.0
        private set
    var totalAmount = 0.0
        private set
    var amountPerPerson = 0.0
        private set
    private val tip_percent: Int
    private val noPeople: Int
    private fun calculate() {
        tipAmount = tip_percent * amount / 100.0
        totalAmount = tipAmount + amount
        amountPerPerson = totalAmount / noPeople
    }

    inner class ZeroAmountException : Exception()

    init {
        if (amount == 0.0) throw ZeroAmountException()
        this.amount = amount
        this.tip_percent = tip_percent
        this.noPeople = noPeople
        calculate()
    }
}